### PR TITLE
Add bouncer tests involving the full deposit monitor pipeline.

### PR DESCRIPTION
**Pull request by MxmUrw**
# Pull Request

Closes: PRO-1773
Closes: PRO-1774

## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [x] I have written sufficient tests.
- [ ] I have written and tested required migrations.
- [ ] I have updated documentation where appropriate.

## Summary

This PR integrates the deposit-monitor (https://github.com/chainflip-io/chainflip-deposit-monitor) into the localnet configuration. The `broker_level_screening` test now uses the monitor to mark transactions as tainted.

### CODE REVIEW

File: bouncer/tests/broker_level_screening.ts [hunk 1]
```diff
@@ -56,36 +55,96 @@ async function newAssetAddress(asset: InternalAsset, seed = null): Promise<strin
...
+        risk_score: { Score: score },
+        unknown_contribution_percentage: 0.0,
+        analysis_provider: 'elliptic_analysis_provider',
+      },
+    ]),
+  });
+
+  return fetch(request).then((res) =>
+    testBrokerLevelScreening.log('got response' + JSON.stringify(res)),
+  );
```
Comment (kylezs) [inline bouncer/tests/broker_level_screening.ts:147]: can probably use the `axios` library instead to remove some of the boilerplate here

File: bouncer/tests/broker_level_screening.ts [hunk 2]
```diff
@@ -122,20 +180,11 @@ async function brokerLevelScreeningTestScenario(
...
-  // Note: The bitcoin core js lib returns the txId in reverse order.
-  // On chain we expect the txId to be in the correct order (like the Bitcoin internal representation).
-  // Because of this we need to reverse the txId before submitting it as tainted.
-  await submitTxAsTainted(hexStringToBytesArray(txId).reverse());
-  await sleep(stopBlockProductionFor);
-  if (stopBlockProductionFor > 0) {
-    pauseBtcBlockProduction(false);
-  }
-  return Promise.resolve(swapParams.channelId.toString());
+  // await setTxRiskScore(txId, 9.0);
```
Comment (kylezs) [inline bouncer/tests/broker_level_screening.ts:185]:
```suggestion
```

File: bouncer/tests/broker_level_screening.ts [hunk 3]
```diff
@@ -173,11 +221,8 @@ async function main() {
   );
   btcRefundAddress = await newAssetAddress('Btc');
 
-  await brokerLevelScreeningTestScenario(
-    '0.2',
-    true,
-    btcRefundAddress,
-    MILLI_SECS_PER_BLOCK * BLOCKS_TO_WAIT,
+  await brokerLevelScreeningTestScenario('0.2', true, btcRefundAddress, 0, (txId) =>
+    setTxRiskScore(txId, 9.0),
```
Comment (kylezs) [inline bouncer/tests/broker_level_screening.ts:225]: seems strange that we always set to the wait time to 0 when we submit with monitor, but it's non-zero when not? I think it makes more sense to encode this in the closure. 
So here we just call it like this (just removing the `waitBeforeReport` parameter) and then below we can do:
```js

 async (txId) => {
      await sleep(MILLI_SECS_PER_BLOCK * 2)
      await submitTxAsTainted(txId);
    },
```

File: localnet/common.sh [hunk 4]
```diff
@@ -172,6 +173,28 @@ build-localnet() {
...
+  $DOCKER_COMPOSE_CMD -f localnet/docker-compose.yml -p "chainflip-localnet" up $CF_CONTAINERS $additional_docker_compose_up_args -d >>$DEBUG_OUTPUT_DESTINATION 2>&1
+  while true; do
+    echo "🩺 Checking deposit-monitor's health ..."
+    REPLY=$(check_endpoint_health 'http://localhost:6060/health')
+    starting=$(echo $REPLY | jq .starting)
+    all_healthy=$(echo $REPLY | jq .all_processors)
+    if test "$starting" == "false" && test "$all_healthy" == "true" ; then
+      echo "💚 deposit-monitor is running!"
+      break
+    fi
```
Comment (kylezs) [inline localnet/common.sh:193]: If we are checking the health here do we need to check it again in the bouncer code?

File: bouncer/tests/broker_level_screening.ts [hunk 5]
```diff
@@ -56,36 +55,96 @@ async function newAssetAddress(asset: InternalAsset, seed = null): Promise<strin
...
-    return false;
+async function ensureDepositMonitorHealth() {
+  const headers: Headers = new Headers();
+  headers.set('Content-Type', 'application/json');
+  headers.set('Accept', 'application/json');
+
+  const request: RequestInfo = new Request('http://127.0.0.1:6060/health', {
+    method: 'GET',
+    headers,
+  });
```
Comment (kylezs) [inline bouncer/tests/broker_level_screening.ts:84]: same here, probably easier to use axios

File: bouncer/tests/broker_level_screening.ts (hunk 1 above)
Comment (MxmUrw) [reply bouncer/tests/broker_level_screening.ts:147]: I'll look into it!

File: bouncer/tests/broker_level_screening.ts (hunk 2 above)
Comment (MxmUrw) [reply bouncer/tests/broker_level_screening.ts:185]: Sure :)

File: localnet/common.sh (hunk 4 above)
Comment (MxmUrw) [reply localnet/common.sh:193]: Yes, if we want to be able to run the BLS test individually, it makes sense to ensure that the deposit-monitor is running. Though if you think better to remove this, sure.

Comment (kylezs) [reply localnet/common.sh:193]: Yeah can see how it's useful, not a strong opinion, happy to leave if you think worthwhile

File: bouncer/tests/broker_level_screening.ts (hunk 1 above)
Comment (MxmUrw) [reply bouncer/tests/broker_level_screening.ts:147]: Done, yes, it helps a bit.

File: bouncer/tests/broker_level_screening.ts (hunk 3 above)
Comment (MxmUrw) [reply bouncer/tests/broker_level_screening.ts:225]: Nice idea, done.

File: localnet/common.sh (hunk 4 above)
Comment (MxmUrw) [reply localnet/common.sh:193]: I left a simpler test in the bouncer code, which does not keep retrying but simply fails. This should cover most scenarios.

File: localnet/common.sh [hunk 6]
```diff
@@ -172,6 +173,28 @@ build-localnet() {
   echo "🤑 Starting LP API ..."
   KEYS_DIR=$KEYS_DIR ./$LOCALNET_INIT_DIR/scripts/start-lp-api.sh $BINARY_ROOT_PATH
 
+  echo "🔬 Starting Deposit Monitor API ..."
+  # On some machines (e.g. MacOS), 172.17.0.1 is not accessible from inside the container, so we need to use host.docker.internal
+  DEPOSIT_MONITOR_ENV='localnet/init/deposit-monitor/deposit-monitor.env'
```
[the remark this answers is not in the document]
Comment (MxmUrw) [reply localnet/common.sh]: Thanks for catching!

File: bouncer/tests/broker_level_screening.ts [hunk 7]
```diff
@@ -56,35 +56,82 @@ async function newAssetAddress(asset: InternalAsset, seed = null): Promise<strin
...
+ * Possible example values are: `"Manual"`, `{Deterministic: {score: 1.0, incomplete_probability: 0.5}}`,
+ * `{Random: { min_score: 0.0, max_score: 10.0 , incomplete_probability: 0.5 }}`.
  */
-function pauseBtcBlockProduction(pause: boolean): boolean {
-  try {
-    execSync(
-      pause
-        ? 'docker exec bitcoin rm /root/mine_blocks'
-        : 'docker exec bitcoin touch /root/mine_blocks',
+async function setMockmode(mode: string | object) {
```
Comment (kylezs) [inline bouncer/tests/broker_level_screening.ts:106]: We could have a type for this that's more specific than "object", also allows typescript to help us a bit more ith type checking

**Review by kylezs — APPROVED** [review]

File: bouncer/tests/broker_level_screening.ts (hunk 7 above)
Comment (MxmUrw) [reply bouncer/tests/broker_level_screening.ts:106]: True, I can create a type for this.

**Review by kylezs — APPROVED** [review]

File: bouncer/tests/broker_level_screening.ts [hunk 8]
```diff
@@ -56,35 +56,89 @@ async function newAssetAddress(asset: InternalAsset, seed = null): Promise<strin
...
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+      timeout: 5000,
+    })
+    .then((res) => res.data)
+    .catch((error) => {
+      let message;
+      if (error.response) {
+        message = error.response.data + ' (' + error.response.status + ')';
```
Comment (kylezs) [inline bouncer/tests/broker_level_screening.ts:91]: nit, format strings make it a bit easier to read (same in the other places too ofc)
```suggestion
message = `${error.response.data} (${error.response.status})`;
```

**Review by kylezs — APPROVED** [review]

File: bouncer/tests/broker_level_screening.ts (hunk 8 above)
Comment (MxmUrw) [reply bouncer/tests/broker_level_screening.ts]: It is indeed nicer this way! Did so.

**Pull request merged**